### PR TITLE
Support loading json files

### PR DESCRIPTION
**Pull request by daeyeon**
IoT.js-DCO-1.0-Signed-off-by: Daeyeon Jeong daeyeon.jeong@samsung.com

### CODE REVIEW

File: src/js/module.js [hunk 1]
```diff
@@ -189,7 +189,15 @@ iotjs_module_t.load = function(id, parent) {
 
   module.filename = modPath;
   module.dirs = [modPath.substring(0, modPath.lastIndexOf('/') + 1)];
-  module.compile();
+
+  var ext = modPath.substr(modPath.lastIndexOf('.') + 1);
+
+  if(ext === 'js') {
```
Comment (LaszloLango) [inline src/js/module.js]: missing space: `if (`

File: src/js/module.js [hunk 2]
```diff
@@ -189,7 +189,15 @@ iotjs_module_t.load = function(id, parent) {
...
+    module.compile();
+  } else if(ext === 'json') {
```
Comment (LaszloLango) [inline src/js/module.js]: ditto: `if (`

File: test/run_pass/test_module_json.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var json = require('./resources/test.json');
+
+assert.equal(json.name, 'IoT.js');
```
Comment (LaszloLango) [inline test/run_pass/test_module_json.js]: I'd add more assert to ensure the whole json file is parsed.

File: src/js/module.js [hunk 4]
```diff
@@ -189,7 +189,15 @@ iotjs_module_t.load = function(id, parent) {
...
+    var source = process.readSource(modPath);
+    module.exports = JSON.parse(source);
```
Comment (andy-cho) [inline src/js/module.js]: Check if it is better to add error/exception handling on JSON parse.
If needed, prepare negative test json entry to test.

Comment (yichoi) [reply src/js/module.js]: A few people are working on it about what you have pointed out.

File: src/js/module.js (hunk 1 above)
Comment (daeyeon) [reply src/js/module.js]: done.

File: src/js/module.js (hunk 2 above)
Comment (daeyeon) [reply src/js/module.js]: done.

File: test/run_pass/test_module_json.js (hunk 3 above)
Comment (daeyeon) [reply test/run_pass/test_module_json.js]: done.

**Comment by yichoi** [comment]
LGTM

**Pull request merged**